Engine/AI/NavMesh.h

```diff
@@ -0,0 +1,190 @@
+#ifndef _NAVMESH_H_
+#define _NAVMESH_H_
+
+#include "AI/SampleDebugDraw.h"
+#include "AI/DebugDrawGL.h"
+#include "Helper/Timer.h"
+
+#include <MathGeoLib.h>
+#include <pcg_basic.h>
+#include <recast/Detour/DetourNavMesh.h>
+#include <recast/Recast/Recast.h>
+#include <vector>
+
+class PanelNavMesh;
+class dtNavMeshQuery;
+class dtNavMesh;
+class dtMeshTile;
+class DetourNavMesh;
+
+
+static const int MAX_POLYS = 256;
+static const int MAX_SMOOTH = 2048;
```

Comment:
This should be inside `NavMesh` definition in order to not contaminate the global context.